### ISSUE TITLE
Sign in using refresh token (React Native)

### ISSUE THREAD

**Issue by elitan** [nhost/nhost#286]
We should allow signing in using a refresh token. This is useful for React Native and OAuth sign-in.

something like:

```
nhost.auth.signIn({refreshToken});
```

**Comment by elitan** [comment]
@Svarto Feel free to add context if needed.

**Comment by Svarto** [comment]
@elitan sure, this will allow for OAuth sign-in within React Native as currently it is possible to leverage the backend OAuth capability in Nhost by creating it through a WebBrowser Authentication session (see code example below). The challenge is that this is done outside the nhost-js SDK, so when a successful login is made and the user is navigated back to the mobile app, the nhost-js SDK is not informed that the user is actually already logged in on the backend. The automatic refresh process of the Access Token is also not initiated.

This suggestion is the fastest way to enable it, as the response from the successful OAuth login includes the refreshToken so the option to login with only a refreshToken will enable Nhost OAuth on RN. 

For reference, this is the response from a successful Nhost backend OAuth login:

```
Object {
  "type": "success",
  "url": "myapp://#refreshToken=xxx",
}```

Google social login example in React Native:
```
export function GoogleSocialLogin() {
  const [result, setResult] = React.useState(null);
  const [request, setRequest] = React.useState(false);

  const _handlePressButtonAsync = async () => {
    setRequest(true);
    let result = await WebBrowser.openAuthSessionAsync(
      "https://xxx.nhost.run/v1/auth/signin/provider/google"
    );
    setResult(result);
    setRequest(false);
  };
  console.log(result);
  return (
    <Button
      disabled={request}
      title="Login"
      onPress={_handlePressButtonAsync}
    />
  );
}```

**Comment by plmercereau** [comment]
Hello, 
With the above PR, you can use `nhost.auth.refreshSession(refreshToken)` to sign in from a refresh token

**Comment by Svarto** [comment]
> Hello, 
> With the above PR, you can use `nhost.auth.refreshSession(refreshToken)` to sign in from a refresh token

Wow, amazing, thank you!

**Comment by Svarto** [comment]
@plmercereau @elitan  I got it to work, will write up a guide soon. However, I encountered a weird case when trying to repeatedly sign in through social login. The returned string with the refreshToken from successfully completing the social login includes two identical refreshToken params.

Note that after the ?, the same refreshToken is returned twice.
`yumee://?refreshToken=dfe3ad07-c3bf-45c9-a8c4-1c48518b7b1a#refreshToken=dfe3ad07-c3bf-45c9-a8c4-1c48518b7b1a`

**Comment by plmercereau** [comment]
Yes, that is something we introduced here
https://github.com/nhost/hasura-auth/pull/146

We initially put the refresh token in the hash part of the url, but it was then not accessible in SSR contexts. So we added to the query parameters. We decided to keep both to make sure hasura-auth is compatible with previous versions of the sdk or with custom-made clients users would come with.
The new versions of the sdk however remove them from both query params and hash after consuming it
Eventually we will stop sending it in the hash.
